utils/land-patterns.stanza

```diff
@@ -9,23 +9,120 @@ defpackage ocdb/land-patterns :
+public defn ref-label (pose:Pose, anchor:Anchor):
+  inside pcb-landpattern:
+    layer(Silkscreen("values", Top)) = Text(">REF", MIN-SILKSCREEN-TEXT-HEIGHT, anchor, pose)
+
+; Creates a reference label at a fixed pose
+public defn ref-label (pose:Pose):
+  ref-label(pose, C)
+
+; DEPRECATED: prefer ref-label(loc(x, y))
+; Creates a reference label at an x,y position and anchor
+public defn ref-label (x:Double, y:Double, anchor:Anchor):
+  ref-label(loc(x, y), anchor)
+
+; DEPRECATED: prefer ref-label(loc(x, y))
+; Creates a reference label at an x,y position with default anchor
 public defn ref-label (x:Double, y:Double) :
   ref-label(x,y,C)
 
+; Create a default reference label
 public defn ref-label () :
-  ref-label(0.0, 0.0)
+  ref-label(loc(0.0, 0.0))
+
+; Create a capsule with normalized rotation
+defn capsule (anchor:Anchor, width:Double, height:Double) -> Capsule:
+  if width >= height:
+    Capsule(anchor, width, height)
+  else:
+    Capsule(width, height, loc(0.0, 0.0, 90.0))
+
+; Create a rounded rectangle with normalized rotation
+defn round-rect (width:Double, height:Double, r:Double) -> RoundedRectangle:
+  if width >= height:
+    RoundedRectangle(width, height, r)
+  else: 
+    RoundedRectangle(height, width, r, loc(0.0, 0.0, 90.0))
+
+; Create a chamfered rectangle with normalized rotation
+defn chamfered (width:Double, height:Double, r:Double) -> ChamferedRectangle:
+  if width >= height:
+    ChamferedRectangle(width, height, r)
+  else:
+    ChamferedRectangle(height, width, r)
+
+;==============================================================================
+;=========================== Geometric Helpers ================================
+;==============================================================================
+; Create a rectangle out of lines
+public defn LineRectangle (w:Double, h:Double, xc:Double, yc:Double, line-w:Double) :
+  val w2 = w * 0.5
+  val h2 = h * 0.5
+  Line(line-w, [
+    Point(xc - w2, yc - h2)
+    Point(xc + w2, yc - h2)
+    Point(xc + w2, yc + h2)
+    Point(xc - w2, yc + h2)
+    Point(xc - w2, yc - h2)])
+
+; Create a rectangle with width, height, center x/y and default line width
+public defn LineRectangle (w:Double, h:Double, xc:Double, yc:Double) :
+  LineRectangle(w, h, xc, yc, MIN-SILKSCREEN-WIDTH)
 
-; ====== Land pattern utilities ===============================================
+; Create a rectangle with width and height and default x/y position
+public defn LineRectangle (w:Double, h:Double) :
+  LineRectangle(w, h, 0.0, 0.0)
+
+; Create a rectangle from dims structure
+public defn LineRectangle (d:Dims): 
+  LineRectangle(x(d), y(d))
+
+; Create a rectangle around another shape
+public defn outline-bounding-box (s:Shape):
+  LineRectangle(dims(s))
+
+; Create a "+" symbol with a fixed length at a center point
+public defn PlusSymbol (center:Point, l:Double) -> Shape:
+  Union $ [
+    Line(MIN-SILKSCREEN-WIDTH, [center + Point(-0.5 * l, 0.0), center + Point(0.5 * l, 0.0)])
```

Comment:
you could also use a `Pose` here (`Pose(center, 0.0, false)`)and apply it to `Point(-0.5 * l, 0.0)` and `Point(0.5 * l, 0.0)`

Reply:
Could I also do `loc(x(center), y(center)) * Union(...)` here?

Reply:
Yes you can. This is a good usage of that.